src/main/java/org/opentripplanner/model/TripPattern.java

```diff
@@ -155,6 +155,35 @@ public void setHopGeometry(int i, LineString hopGeometry) {
         this.hopGeometries[i] = CompactLineString.compactLineString(hopGeometry,false);
     }
 
+    /**
+     * This will copy the geometry from another TripPattern to this one. It checks if each hop is
+     * between the same stops before copying that hop geometry. If the stops are different, a
+     * straight-line hop-geometry will be used instead.
+     *
+     * @param other TripPattern to copy geometry from
+     */
+    public void setHopGeometriesFromPreviousTripPattern(TripPattern other) {
```

Comment:
rename this method, "previous" has a time or sequence connotation. The method is general enough that it could be setHopGeometriesFromTripPattern.